recipes-soletta/soletta/soletta_git.bb

```diff
@@ -93,6 +199,13 @@ do_configure_append() {
    # becoming invalid
    # Also, yocto has a toolchain that will treat RPATH for Soletta
    sed -i "s/^RPATH=y/# RPATH is not set/g" ${S}/.config
+
+   if ${@bb.utils.contains('DISTRO_FEATURES', 'x11', 'true', 'false', d)} == 'true'; then
+       if [ -n "$(cat ${S}/.config | grep '^# HAVE_GTK is not set')" ]; then
+          sed -i "s/^# HAVE_GTK is not set/HAVE_GTK=y/g" ${S}/.config
```

Comment:
I would keep PACKAGEFEATURE as it was before (i.e. enable x11 package feature by default iff distro x11 feature is on), and then add this .config mangling depending on PACKAGEFEATURE.

That makes the recipe a bit more versatile: non-x11 soletta in a x11 distro would become possible.